client/components/social-buttons/google.js

```diff
@@ -71,6 +73,21 @@ class GoogleLoginButton extends Component {
 		return window.gapi;
 	}
 
+	async initializeAuth2( gapi ) {
```

Comment:
Curious as I haven't contributed to calypso for a while, how do we feel about mixing promises and async/await?

Reply:
@Tug `asunc/await` is essentially just syntactic sugar around promises, so I tend to use them both, depending on which one comes more naturally to me for a particular piece of code.

I don't think there's any rule against any one of them (or against using them both in the same file) in Calypso, but I'm always happy to rewrite my code if folks find it confusing!

Reply:
I think that the big `then` chain in the `initialize` function would benefit from a rewrite to async/await. For example, passing the `gapi` value around with calls like `x.then( () => gapi )` is a typical thing that looks weird with promises and is much nicer with async/await.

But that's for another PR -- this patch is quite urgent and it's already a shame that it took a month to merge.